### PR TITLE
chore(deps): update overlay

### PR DESCRIPTION
**Pull request by cutterbl**
Update the react-overlay dependency and usage within the project. Necessary update to avoid conflicts with those using latest version of react-overlay (for instance, those creating Modal managers in their projects, consuming react-bootstrap).

### CODE REVIEW

**Comment by arecvlohe** [comment]
@jquense Thoughts?

**Comment by cutterbl** [comment]
I fixed the merge conflict. Since the Popper get's it's `ref` from Overlay, there's no need for the `createRef` directly in the Popper.

**Comment by cutterbl** [comment]
Sorry for the back and forth. The only OS projects I regularly send updates to are my own, so reacquainting with the workflow here. Think all the bits are there now. Tests are passing, examples are rebuilt and running properly, and no console messages.

File: src/Month.js [hunk 1]
```diff
@@ -12,7 +12,7 @@ import getPosition from 'dom-helpers/query/position'
 import raf from 'dom-helpers/util/requestAnimationFrame'
 
 import Popup from './Popup'
-import Overlay from 'react-overlays/lib/Overlay'
+import { Overlay } from 'react-overlays'
```
Comment (jquense) [inline src/Month.js]: lets keep the cherry picked import i don't want to rely on treeshaking working correctly if we don't need to

Comment (jquense) [reply src/Month.js]: `import Overlay from 'react-overlays/Overlay'`

Comment (cutterbl) [reply src/Month.js]: Done

**Comment by jquense** [comment]
thanks!

**Pull request merged**